### PR TITLE
chore: replace querystring and query-string with URLSearchParams

### PR DESCRIPTION
**Pull request by ashtonG**
## Description

This removes the `querystring` and `query-string` libraries and replaces them with the native URLSearchParams. While I was in the bindnigs, I also replaced `portable-fetch` with the native window fetch. We still use `whatwg-fetch` for the test environment so it's still installed here.

## Test Plan

* unit tests should still work
* get requests should still work

## Commentary (optional)

handling of arrays is different between `URLSearchParams` and the other libraries, so for params where the type is unknown, we get all values for the key and _iff_ the resulting array is length 1, we use the single value.

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-1076

### CODE REVIEW

**Review by keita-determined — COMMENTED** [review]
looks like experiment table on netlify doest get any data but it does on other PRs. I assume the PR caused it. can you take a look?
also could you have @hkang1 review to double-check? useSettings tends to cause some issues.
nice branch name btw.

File: webui/react/src/ee/SamlAuth.ts [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-  router.getRouter().navigate(`${window.location.pathname}?${queryString.stringify(newQueries)}`);
+  const relayState = new URLSearchParams(clone.get('relayState') || '');
+  for (const key of relayState.keys()) {
+    // not using entries here in order to handle arrays properly
+    const value = relayState.getAll(key);
+    clone.set(key, value[0]);
+    value.slice(1).forEach((subValue) => clone.append(key, subValue));
+  }
+  clone.delete('relayState');
+  router.getRouter().navigate(`${window.location.pathname}?${clone}`);
```
[the remark this answers is not in the document]
Comment (ashtonG) [reply webui/react/src/ee/SamlAuth.ts:21]: we could but that'd require us to pass the location in from the callsite.

**Review by hkang1 — COMMENTED** [review]
Thanks for taking this on!

I pulled the branch down and built it to give it a shot locally but I noticed there was an issue with experiment list. The experiment list responds with 400 Bad Requests but cluster would be fine and report ok. Login/Logout, Tasks, Master Logs, Model Registry (couldn't try detail views because I couldn't access any experiments) and Workspaces/Projects look ok as well. Haven't been able to try everything out, but so far it seems that there are some minor issues around experiment endpoints.

File: webui/react/src/hooks/useAuthCheck.ts [hunk 2]
```diff
@@ -12,7 +11,7 @@ import { getCookie } from 'utils/browser';
 
 const useAuthCheck = (): (() => void) => {
   const info = useObservable(determinedStore.info);
-  const location = useLocation();
+  const [searchParams] = useSearchParams();
```
[the remark this answers is not in the document]
Comment (ashtonG) [reply webui/react/src/hooks/useAuthCheck.ts:14]: The semantics of `useSearchParams` and `new URLSearchParams` are different. I tried to keep the same usage (`useSearchParams` for places where we're using the router location and only looking at search parameters, `new URLSearchParams` for places where we're using the window location OR we're using `useLocation` for something else) in order to keep the surface area of the pr small. If you want to standardize usage, I think that'd be a good candidate for a ticket?

In addition, I think setting search parameters should push the url and not replace it. We should really only replace the history entry if it's invalid somehow (i.e. redirecting from an old url to a new url)

**Review by keita-determined — COMMENTED** [review]
thanks for the fix. exp table is visible now, but APIs are called many times in exp table.
Also, there's merge conflict

https://user-images.githubusercontent.com/109113616/235500512-3b99e4b8-4604-4e07-9a48-4dddffe30c93.mov

**Comment by ashtonG** [comment]
@keita-determined i'm seeing that on latest master, so I don't think that's due to these changes and i just rebased to clear the merge conflict.

**Review by hkang1 — APPROVED** [review]
awesome, this is good to go! experiments is functioning as expected

**Pull request merged**